res/controllers/Hercules-DJControl-Inpulse-500-script.js

```diff
@@ -0,0 +1,1566 @@
+  if (value > 0x3F) {
+    if (DJCi500.browserOffFocusMode) {
+      engine.setValue('[Playlist]', 'SelectTrackKnob', -1);
+    } else {
+      engine.setValue('[Library]', 'MoveUp', 1);
+    }
+  } else {
+    if (DJCi500.browserOffFocusMode) {
+      engine.setValue('[Playlist]', 'SelectTrackKnob', 1);
+    } else {
+      engine.setValue('[Library]', 'MoveDown', 1);
+    }
+  }
+}
+
+DJCi500.spinback_button = function(channel, control, value, status, group) {
+  var deck = parseInt(group.substring(8,9)); // work out which deck we are using
+  engine.spinback(deck, value > 0, 2.5); // use default starting rate of -10 but decrease speed more quickly
+}
+
+// Update the Tempo and phase sync leds
+DJCi500.tempoLEDs = function () {
+  // Current active decks
+  var deckA = DJCi500.deckA.currentDeck;
+  var deckB = DJCi500.deckB.currentDeck;
+
+  //Tempo:
+  var tempo1 = engine.getValue(deckA, "bpm");
+  var tempo2 = engine.getValue(deckB, "bpm");
+  var diff = tempo1 - tempo2;
+
+  //Check double tempo:
+  var doubleTempo = 0;
+  if (diff > 0){
+    if ((tempo1 / tempo2) > 1.5){doubleTempo = 1; diff = tempo1/2 - tempo2;}
+  }
+  else{
+    if ((tempo2 / tempo1) > 1.5){doubleTempo = 1; diff = tempo1 - tempo2/2;}
+  }
+
+  if ( diff < -0.25)
+  {
+    //Deck1
+    midi.sendShortMsg(0x91, 0x1E, 0x0);
+    midi.sendShortMsg(0x91, 0x1F, 0x7F);
+    midi.sendShortMsg(0x91, 0x2C, 0x0);
+    //Deck2
+    midi.sendShortMsg(0x92, 0x1F, 0x0);
+    midi.sendShortMsg(0x92, 0x1E, 0x7F);
+    midi.sendShortMsg(0x92, 0x2C, 0x0);
+
+    //clear beatalign leds
+    //Deck1
+    midi.sendShortMsg(0x91, 0x1C, 0x0);
+    midi.sendShortMsg(0x91, 0x1D, 0x0);
+    midi.sendShortMsg(0x91, 0x2D, 0x0);
+    //Deck2
+    midi.sendShortMsg(0x92, 0x1C, 0x0);
+    midi.sendShortMsg(0x92, 0x1D, 0x0);
+    midi.sendShortMsg(0x92, 0x2D, 0x0);
+  }
+  else if ( diff > 0.25)
+  {
+    //Deck1
+    midi.sendShortMsg(0x91, 0x1F, 0x0);
+    midi.sendShortMsg(0x91, 0x1E, 0x7F);
+    midi.sendShortMsg(0x91, 0x2C, 0x0);
+    //Deck2
+    midi.sendShortMsg(0x92, 0x1E, 0x0);
+    midi.sendShortMsg(0x92, 0x1F, 0x7F);
+    midi.sendShortMsg(0x92, 0x2C, 0x0);
+
+    //clear beatalign leds
+    //Deck1
+    midi.sendShortMsg(0x91, 0x1C, 0x0);
+    midi.sendShortMsg(0x91, 0x1D, 0x0);
+    midi.sendShortMsg(0x91, 0x2D, 0x0);
+    //Deck2
+    midi.sendShortMsg(0x92, 0x1C, 0x0);
+    midi.sendShortMsg(0x92, 0x1D, 0x0);
+    midi.sendShortMsg(0x92, 0x2D, 0x0);
+  }
+  else {
+    //Deck1
+    midi.sendShortMsg(0x91, 0x1E, 0x0);
+    midi.sendShortMsg(0x91, 0x1F, 0x0);
+    midi.sendShortMsg(0x91, 0x2C, 0x7F);
+    //Deck2
+    midi.sendShortMsg(0x92, 0x1E, 0x0);
+    midi.sendShortMsg(0x92, 0x1F, 0x0);
+    midi.sendShortMsg(0x92, 0x2C, 0x7F);
+
+    //Do beat alignement only if the tracks are already on Tempo
+    // and only if they are playing
+    if ( (engine.getValue(deckA, "play_latched")) && (engine.getValue(deckB, "play_latched")) ){

```

Comment:
Formatting nitpick: Consistent whitespace should be used, unnecessary parentheses can be omitted.
```suggestion
    if (engine.getValue(deckA, "play_latched") && engine.getValue(deckB, "play_latched")){

```